tonic/functional/event_downsampling.py

```diff
@@ -0,0 +1,271 @@
+import numpy as np
+from numpy.lib.recfunctions import unstructured_to_structured
+
+def naive_downsample(events: np.ndarray, sensor_size: tuple, target_size: tuple):
+    """Downsample the classic "naive" Tonic way. Multiply x/y values by a spatial_factor 
+    obtained by dividing sensor size by the target size.
+    Parameters:
+        events (ndarray): ndarray of shape [num_events, num_event_channels].
+        sensor_size (tuple): a 3-tuple of x,y,p for sensor_size.
+        target_size (tuple): a 2-tuple of x,y denoting new down-sampled size for events to be
+                             re-scaled to (new_width, new_height).
+    Returns:
+        the downsampled input events.
+    """
+    
+    assert "x" and "y" in events.dtype.names
+    
+    events = events.copy()
+    
+    spatial_factor = np.asarray(target_size) / sensor_size[:-1]
+
+    events["x"] = events["x"] * spatial_factor[1]
+    events["y"] = events["y"] * spatial_factor[0]
+
+    return events
+
+def time_bin_numpy(events: np.ndarray, time_bin_interval: float):
```

Comment:
can you reuse tonic.slicers.slice_events_by_time instead of that? see for example here https://github.com/neuromorphs/tonic/blob/develop/tonic/functional/to_timesurface.py#L33

Reply:
Done